core-tests/src/test/java/org/mule/runtime/core/internal/streaming/bytes/PoolingByteBufferManagerTestCase.java

```diff
@@ -109,22 +108,22 @@ public void limitTotalMemory() throws Exception {
     bufferManager.dispose();
     bufferManager = new PoolingByteBufferManager(allocateExecutor, memoryManager, waitTimeoutMillis);
 
-    assertMemoryLimit(bufferCapacity, waitTimeoutMillis);
+    assertMemoryLimit(bufferCapacity.intValue(), waitTimeoutMillis);
   }
 
   @Test
   public void limitTotalMemoryThroughSystemProperty() throws Exception {
     final long maxMemory = 40;
-    final int bufferCapacity = toIntExact(maxMemory / 4);
+    final Long bufferCapacity = round(maxMemory * MAX_STREAMING_PERCENTILE) / 2;
     final long waitTimeoutMillis = SECONDS.toMillis(2);
 
     MemoryManager memoryManager = mock(MemoryManager.class);
 
     bufferManager.dispose();
-    setProperty(MULE_STREAMING_MAX_MEMORY, String.valueOf(maxMemory / 2));
+    setProperty(MULE_STREAMING_MAX_MEMORY, String.valueOf(new Double(maxMemory * MAX_STREAMING_PERCENTILE).intValue()));
```

Comment:
shouldn't this be `longValue` instead of `intValue`?

Reply:
nope